### ISSUE TITLE
gpb doesn't build on windows/cygwin

### ISSUE THREAD

**Issue by bb4242** [tomas-abrahamsson/gpb#31]
Trying to build gpb on windows (from a cygwin shell) fails:
```
$ rebar get compile
==> gpb (get-deps)
==> gpb (compile)
'build' is not recognized as an internal or external command,
operable program or batch file.
ERROR: Command [compile] failed!
```
This seems like a rebar issue; I'll file a separate bug with that project.

**Comment by tomas-abrahamsson** [comment]
I've just pushed a workaround to a branch, _pre-post-hooks-on-cygwin_, based on your findings in rebar/rebar#450. It appears to work on Linux, but it would be nice if you would be able to try it on windows/cygwin to give it a go or no go.

**Comment by tomas-abrahamsson** [comment]
FYI: I just updated the branch, with a minor fix.

**Comment by bb4242** [comment]
Cool, thanks!  I noticed one problem with the branch on cygwin: the `pwd` causes problems because my Erlang installation doesn't know how to access absolute cygwin paths like `/home/user/gpb/descr_src/gpb_descriptor.proto`.  I found that I can make it work by changing over to relative paths (see pull request https://github.com/tomas-abrahamsson/gpb/pull/33).  That seems to work for me on both linux and cygwin, but please let me know if there are any implications of moving away from absolute paths.

**Comment by bb4242** [comment]
As a side note, rebar still has a problem on cygwin even after gpb compiles correctly: it can't seem to find and use gpb to compile a project that contains proto files, failing with this error:
```
ERROR: Protobuffs library not present in code path!
```
I get this even though I have `{proto_compiler, gpb}` in my rebar.config (which works fine on linux).  It might warrant opening a separate ticket for that issue, though.

**Comment by tomas-abrahamsson** [comment]
Just merged #33, so closing this. I think the rebar issue is somehow code path related. Does your `$ERL_LIBS` contain a directory that contains gpb? (ie the gpb directory is not supposed to be directly in `$ERL_LIBS`, but rather it is supposed to contain a directory where erlang apps, such as gpb, are located.  Reading the source for `code_server.erl`, it seems on windows `$ERL_LIBS` is a semicolon-separated list of directories, on other platforms, it is colon-separated.

**Comment by bb4242** [comment]
Thanks for merging https://github.com/tomas-abrahamsson/gpb/pull/33 - that's great.  I'll do some more experimentation with the rebar issue.  On further inspection, it looks like it actually is finding gpb.  Doing a `rebar compile -vv` on cygwin from a project that has proto files gives:
```
==> systemmessages (compile)
DEBUG: Matched required ERTS version: 6.3 -> .*
DEBUG: Matched required OTP release: 17 -> .*
DEBUG: Min OTP version unconfigured
DEBUG: is_app_available, looking for App gpb with Path "c:/cygwin/home/bbethke/systemmessages/deps/gpb"
INFO:  Looking for gpb- ; found gpb-3.17.10 at c:/cygwin/home/bbethke/systemmessages/deps/gpb
ERROR: Protobuffs library not present in code path!
ERROR: compile failed while processing C:/cygwin/home/bbethke/systemmessages: rebar_abort
```
So it looks like it finds gpb, but still thinks it should be compiling with protobuffs (despite having `{proto_compiler, gpb}.` in rebar.config.  I'll post this info in https://github.com/rebar/rebar/issues/450 as well.
